literature_search.py

```diff
@@ -0,0 +1,298 @@
+                          'journal': journals})
+
+for index, row in papers_df.iterrows():
+    if row['abstract'] == '' or row['author'] == 'AUTHOR NAMES ERROR' or row['title'] == '':
+        papers_df.drop(index,inplace=True)
+
+#join titles, keywords, and abstract
+papers_df['everything'] = pd.DataFrame(papers_df['title'].astype(str)*4+\
+                                       papers_df['abstract'].astype(str)+\
+                                       papers_df['keywords'].astype(str))
+
+titles_vec = vect.transform(papers_df['everything'])
+# OR if you don't want to use just the title:
+# titles_vec = vect.transform(papers_df['title'])
+
+#==============================================================================
+#========================= Predict topics =====================================
+# Predict Topics For Each Paper
+prediction_vec = model.predict(titles_vec)
+
+topics = []
+pred_val = []
+title_temp = []
+indx = []
+
+for k, top_val in enumerate(prediction_vec):
+    try:
+        papers_df.loc[k,:]
+        pred_val = np.max(top_val)
+        if pred_val > 0*np.sort(top_val)[-2]:
+            indx.append(k)
+            topics.append(le.inverse_transform([np.argmax(top_val)])[0])
+            title_temp.append(papers_df['title'][k])
+        else:
+            indx.append(k)
+            topics.append('unknown')
+            title_temp.append(papers_df['title'][k])
+    except:
+        print('Skipping prediction of paper #: ' + str(k))
+papers_df = pd.DataFrame(data = {'title': title_temp,
+                                  'topic': topics})
+
+#==============================================================================
+#========================= Save Titles and Topics =============================
+
+#add info for github markdown format
+papers_df['title']      = [title if title[1] is not '[' else title[1:-1] for title in papers_df['title']]
+papers_df['authors']    = [authors[k] if authors[k][1] is not '[' else authors[1:-1] for k in indx]
+papers_df['journal']    = [journals[k] for k in indx]
+papers_df['links']      = [links[k] for k in indx]
+papers_df['full_title'] = [full_titles[k] for k in indx]
+#generate filename
+import datetime
+now = datetime.datetime.now()
+strings = [str(now.year), str(now.month), str(now.day),'litupdate.csv']
+fname = 'Literature_Updates/'+'-'.join(strings)
+strings = [str(now.year), str(now.month), str(now.day),'litupdate.md']
+mdname = 'Literature_Updates/'+'-'.join(strings)
+strings = [str(now.year), str(now.month), str(now.day),'litupdate']
+urlname = '-'.join(strings)
+
+print('Filename: ',fname)
+
+# Compare to previously searched papers
+#old_papers = pd.concat([pd.read_csv('Literature_Updates/2019-5-7-litupdate.csv'),
+#                        pd.read_csv('Literature_Updates/2019-4-30-litupdate.csv')])
+compare_papers = pd.read_csv('Literature_Updates/compare_papers.csv')
+
+# Probably Don't need to compare against training data, could remove for speed?
```

Comment:
I agree that this can probably be commented out. We don't really get duplicates with our current method of searching PubMed and I think it's unlikely that we'll get repeats from 2008.

Reply:
I'm thinking for now we should leave it in, as until they stop posting the weekly update we should compare it to the training data. It probably won't ever happen, but its not like its that slow. 

A lot of these when I say remove for speed will probably only apply if we had a billion jillion papers, and won't actually need to be fixed/altered.